configure.ac

```diff
@@ -1946,6 +1947,9 @@ CPPFLAGS_TEMP="$CPPFLAGS"
 unset CPPFLAGS
 CPPFLAGS="$CPPFLAGS_TEMP"
 
+if test -n "$use_sanitizers"; then
+  export SECP_CFLAGS="$SECP_CFLAGS $SANITIZER_CFLAGS"
```

Comment:
My take on `SECP_CFLAGS` is that it's an internal of the libsecp256k1 build system, and users are not supposed to set it (and are not guaranteed that it does anything meaningful). The proper user variable is just `CFLAGS`. 

You could either export `CFLAGS="-g $SANITIZER_CFLAGS"` temporarily, or perhaps add a CFLAGS arg to `ac_configure_args`.

We could also expose the `SECP_CFLAGS` "officially" in the lib. Then it should probably be renamed to `SECP256K1_CFLAGS`. But I think it's a bit arbitrary, given there's an official way to set CFLAGS (namely setting CFLAGS ;)).

There's also this, which may be an elegant solution or overkill, depending on your taste... https://www.gnu.org/software/autoconf-archive/ax_subdirs_configure.html

Sorry for noticing this only now in a second review...

Reply:
I was thinking about the best way to handle this, and my current take was that given this is only when we are using sanitizers, and we are a somewhat non-standard/more-tightly-coupled users of secp in any case, I wasn't too worried. Happy for any other approach that achieves the same thing, but also don't want secp to expose additional (non-standard) flags just for us.

Reply:
> I wasn't too worried.

Yep, I agree, any approach is ok in the end. My feeling is just that setting `SECP_CFLAGS` is not the cleanest choice and more likely than my other suggestions to cause breaks in the future.

Reply:
> My feeling is just that setting SECP_CFLAGS is not the cleanest choice and more likely than my other suggestions to cause breaks in the future.

We can accept that possibility for now, and at the next subtree update (if/when things break), we'll take another look.